src/adapter.rs

```diff
@@ -67,8 +67,9 @@ macro_rules! offset_of {
         let $container { $field: _, .. };
 
         // Create an instance of the container and calculate the offset to its
-        // field. Although we are creating references to uninitialized data this
-        // is fine since we are not dereferencing them.
+        // field. We are creating references to uninitialized data, which is
+        // not allowed and UB under the current rules---but offset_of cannot
+        // currently be implemented without it.
         #[allow(unused_unsafe)]
         let val: $container = unsafe { $crate::__core::mem::uninitialized() };
```

Comment:
Also note that this version of the macro will panic if the `container` type has an uninhabited field: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=27609b4b89247aaa4ef29504717a06e2.

Reply:
I'm aware, but there's nothing we can do about it without `MaybeUninit`.

Reply:
Well there are trade-offs here. [This version](https://github.com/Gilnaa/memoffset/blob/fa2cce2358afd48f40ed1fa7229c7a29bf511ad2/src/offset_of.rs#L67) uses a dangling pointer instead of an uninitialized local. My gut feeling is that that's less likely to cause problems. It also solves the issue with panics.